crates/uv-fs/src/lib.rs

```diff
@@ -556,7 +556,16 @@ impl LockedFile {
         path: impl AsRef<Path>,
         resource: impl Display,
     ) -> Result<Self, std::io::Error> {
-        let file = fs_err::File::create(path.as_ref())?;
+        #[allow(unsafe_code)]
+        #[cfg(unix)]
+        let old_umask = unsafe { nix::libc::umask(0) }; // for multi user file locking
+        let file = fs_err::File::create(path.as_ref());
+        #[cfg(unix)]
+        #[allow(unsafe_code)]
+        unsafe {
+            nix::libc::umask(old_umask);
+        };
```

Comment:
Can you use [`nix::sys::stat::umask`](https://docs.rs/nix/latest/nix/sys/stat/fn.umask.html#) here instead? (Otherwise, you're adding a dependency here on `nix` when just `libc` would do. But if we're depending on `nix`, we might as well use the safe API...)

Reply:
> Can you use [`nix::sys::stat::umask`](https://docs.rs/nix/latest/nix/sys/stat/fn.umask.html#) here instead? (Otherwise, you're adding a dependency here on `nix` when just `libc` would do. But if we're depending on `nix`, we might as well use the safe API...)

Tested, `nix::sys::stat::umask` works and is safe.